Package.swift

```diff
@@ -303,9 +303,13 @@ let package = Package(
         .executableTarget(
             name: "SmithyCodegenCLI",
             dependencies: [
+                "SmithyCodegenCore",
                 .product(name: "ArgumentParser", package: "swift-argument-parser"),
             ]
         ),
+        .target(
+            name: "SmithyCodegenCore"
+        ),
```

Comment:
Created this target for codegen components.

Eventually this may be exposed as a library, but for now it's solely a `smithy-swift` internal target.